services/system-user-service/src/test/java/com/amazon/aws/partners/saasfactory/saasboost/keycloak/KeycloakTestUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazon.aws.partners.saasfactory.saasboost.keycloak;
+
+import com.amazon.aws.partners.saasfactory.saasboost.SystemUser;
+import org.keycloak.representations.idm.UserRepresentation;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+public final class KeycloakTestUtils {
+    public static void assertUserListsEqual(List<UserRepresentation> expected, List<UserRepresentation> actual) {
+        // if expected and actual are not equal in existence
+        if (expected == null ^ actual == null) {
```

Comment:
Wow XOR for the win!
Replace with `((expected == null && actual == null) || (expected != null && actual != null))` for clarity?

Reply:
> Any sense in incorporating resilience4j retry in these 3rd party calls to REST APIs? Like we did in CloudFormationUtils?

I like the idea in concept, but what would we retry? The only thing I can think of would be `429 Too Many Requests`, since we have no indication that [any other HTTP code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors) would necessarily be retryable. In addition I can find no documentation or code reference that indicates a throttling mechanism in Keycloak.